amethyst_renderer/src/pass/flat.rs

```diff
@@ -67,7 +68,10 @@ where
     N: Component + AsRef<Material> + Send + Sync,
 {
     type Data = (
-        Option<Fetch<'a, Camera>>,
+        Fetch<'a, Orientation>,
+        ReadStorage<'a, ActiveCamera>,
+        ReadStorage<'a, Camera>,
+        ReadStorage<'a, LocalTransform>,
```

Comment:
Shouldn't this use `Transform` to build the view matrix, as opposed to `LocalTransform`? What if the camera gets attached to another object?

Reply:
The camera stores its translation and rotation in a `LocalTransform`. That's per definition. If the camera does not have a `LocalTransform` components, the join further down will return `None`, which means you do not have a camera.

Reply:
That shouldn't be the case at all, and I don't see how it is the case here either. 

`LocalTransform` holds the transform of the entity relative to the transform of the parent. `Transform` holds the absolute transformation of an entity. Entities are able to function without `LocalTransform` just fine.